experiments/bandwidth_accounting/post_process_bandwidth_accounting.py

```diff
@@ -0,0 +1,51 @@
+#!/usr/bin/env python3
+import os
+import sys
+from pathlib import Path
+
+from pony.orm import db_session
+
+from tribler_core.modules.bandwidth_accounting.database import BandwidthDatabase
+from tribler_core.modules.bandwidth_accounting.transaction import BandwidthTransactionData
+
+from gumby.statsparser import StatisticsParser
+
+
+class BandwidthStatisticsParser(StatisticsParser):
+    """
+    Parse TrustChain statistics after an experiment has been completed.
+    """
+
+    def __init__(self, node_directory):
+        super().__init__(node_directory)
+        self.output_dir = os.path.join(os.environ['PROJECT_DIR'], 'output')
+
+    def aggregate_databases(self):
+        transactions = []
+
+        for _, filename, _ in self.yield_files('bandwidth.db'):
+            print("Considering database %s" % os.path.abspath(filename))
+            db = BandwidthDatabase(Path(os.path.abspath(filename)), None)
+            with db_session:
+                db_txs = list(db.BandwidthTransaction.select())
+                for db_tx in db_txs:
+                    tx = BandwidthTransactionData.from_db(db_tx)
+                    transactions.append(tx)
+
+        print("Found %d transactions..." % len(transactions))
+        all_payouts_db = BandwidthDatabase(Path(self.output_dir) / "bandwidth.db", None)
+        for tx in transactions:
+            all_payouts_db.BandwidthTransaction.insert(tx)
+
+        with open("total_payout_edges.txt", "w") as out_file:
+            out_file.write("%d" % len(transactions))
+
+    def run(self):
+        self.aggregate_databases()
+
+
+# cd to the output directory
+os.chdir(os.environ['OUTPUT_DIR'])
```

Comment:
This is very error-prone. I would suggest stripping this and using `with open(os.path.join(os.environ['OUTPUT_DIR'], "total_payout_edges.txt"), "w") as out_file:` on line 40 instead.

I assume that is the only reason this `chdir` is here, please let me know if that is not the case and I will update my review.

Reply:
I had to dig a bit deeper into this, but it seems that the argument passed to the init method of `BandwidthStatisticsParser` is `.`. Therefore, this class makes the hidden assumption that the current working directory is in the experiment output directory, which is definitely not a good design. However, this code has been in Gumby for many, many years now. Fixing this is possible but would require refactoring efforts unrelated to this PR. Therefore, I opt to leave it like this for now. 👍

Reply:
Seems reasonable, let's defer that to an issue.